src/network/network_gui.cpp

```diff
@@ -1675,38 +1675,33 @@ struct NetworkClientListWindow : Window {
 		this->vscroll->SetCapacityFromWidget(this, WID_CL_MATRIX);
 	}
 
-	void SetStringParameters(WidgetID widget) const override
+	std::string GetWidgetString(WidgetID widget, StringID stringid) const override
 	{
 		switch (widget) {
 			case WID_CL_SERVER_NAME:
-				SetDParamStr(0, _network_server ? _settings_client.network.server_name : _network_server_name);
-				break;
+				return GetString(stringid, _network_server ? _settings_client.network.server_name : _network_server_name);
 
 			case WID_CL_SERVER_VISIBILITY:
-				SetDParam(0, STR_NETWORK_SERVER_VISIBILITY_LOCAL + _settings_client.network.server_game_type);
-				break;
+				return GetString(stringid, STR_NETWORK_SERVER_VISIBILITY_LOCAL + _settings_client.network.server_game_type);
 
 			case WID_CL_SERVER_INVITE_CODE: {
 				static std::string empty = {};
-				SetDParamStr(0, _network_server_connection_type == CONNECTION_TYPE_UNKNOWN ? empty : _network_server_invite_code);
-				break;
+				return GetString(stringid, _network_server_connection_type == CONNECTION_TYPE_UNKNOWN ? empty : _network_server_invite_code);
```

Comment:
The `empty` doesn't need to be a static variable any more since we pass them by-value now.